### PR TITLE
[MI-2963]: Did the review fixes of gitlab PR #301

### PR DESCRIPTION
**Pull request by Nityanand13**
[PR: #301](https://github.com/mattermost/mattermost-plugin-gitlab/pull/301)

### CODE REVIEW

File: webapp/src/components/link_tooltip/link_tooltip.test.jsx [hunk 1]
```diff
@@ -1,43 +1,67 @@
...
-        const expected = [''];
+        const expected = {
+            owner: '',
+        };
 
         const result = getInfoAboutLink(href, hostname);
+        expect(result).toEqual(expected);
+    });
+
+    it('Should return an object invalid owner, repo, type, and number when given a valid hostname and invalid href', () => {
```
Comment (raghavaggarwal2308) [inline webapp/src/components/link_tooltip/link_tooltip.test.jsx:54]: Add a testcase for issue comment as well as mentioned in the [comment](https://github.com/mattermost/mattermost-plugin-gitlab/pull/301#discussion_r1153742764)

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]
Look at [this comment](https://github.com/mattermost/mattermost-plugin-gitlab/pull/301#discussion_r1153737570) as well.

In the default case just log a message example: "this link type is not supported" and break from the switch case

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 2]
```diff
@@ -24,10 +29,19 @@ const LINK_TYPES = {
     ISSUES: 'issues',
 };
 
-export const getInfoAboutLink = (href, hostname) => href.split(`${hostname}/`)[1].split('/');
+export const getInfoAboutLink = (href, hostname) => {
+    const linkInfo = href.split(`${hostname}/`)[1].split('/');
+    return {
```
Comment (raghavaggarwal2308) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:34]: Add a check if length is greater than or equal to 5

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 3]
```diff
@@ -82,6 +96,8 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
             color = data.state === STATE_TYPES.OPENED ? OPENED_COLOR : CLOSED_COLOR;
             iconType = data.state === STATE_TYPES.OPENED ? IssueOpenedIcon : IssueClosedIcon;
             break;
+        default:
+            dispatch(logError('this link type is not supported'));
```
Comment (raghavaggarwal2308) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:100]:
```suggestion
            dispatch(logError('this link type is not supported to display a tooltip'));
```

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 4]
```diff
@@ -36,19 +50,19 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
-                switch (type) {
+                switch (linkInfo.type) {
                 case LINK_TYPES.ISSUES:
-                    res = await Client.getIssue(owner, repo, number);
+                    res = await Client.getIssue(linkInfo.owner, linkInfo.repo, linkInfo.number);
                     break;
                 case LINK_TYPES.MERGE_REQUESTS:
-                    res = await Client.getPullRequest(owner, repo, number);
+                    res = await Client.getPullRequest(linkInfo.owner, linkInfo.repo, linkInfo.number);
                     break;
```
Comment (raghavaggarwal2308) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:61]: add the default case here as well

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 5]
```diff
@@ -36,19 +53,21 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
+                default:
+                    dispatch(logError('this link type is not supported to display a tooltip'));
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:66]: make this warn

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 6]
```diff
@@ -82,6 +101,8 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
             color = data.state === STATE_TYPES.OPENED ? OPENED_COLOR : CLOSED_COLOR;
             iconType = data.state === STATE_TYPES.OPENED ? IssueOpenedIcon : IssueClosedIcon;
             break;
+        default:
+            dispatch(logError('this link type is not supported to display a tooltip'));
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:105]: same

Comment (Kshitij-Katiyar) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:105]: do something like and see if this work or explore
```suggestion
            dispatch(logError('link type %s is not supported to display a tooltip ',data.type) );
```

File: webapp/src/components/link_tooltip/link_tooltip.test.jsx [hunk 7]
```diff
@@ -1,43 +1,63 @@
...
-        const expected = ['mattermost', 'test', '-', 'merge_requests'];
 
         const result = getInfoAboutLink(href, hostname);
-
-        expect(result).toEqual(expected);
+        expect(result).toEqual({});
     });
 
-    it('Should return the correct owner, repo, undefined type, and number when given a valid GitLab href and hostname missing the type', () => {
+    it('Should return an empty object when given a valid GitLab hostname and href missing the type', () => {
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/link_tooltip/link_tooltip.test.jsx:26]: Should return an empty object when given a valid GitLab hostname and href with missing the type parameter

File: webapp/src/components/link_tooltip/link_tooltip.jsx (hunk 5 above)
Comment (Nityanand13) [reply webapp/src/components/link_tooltip/link_tooltip.jsx]: Discussed

File: webapp/src/components/link_tooltip/link_tooltip.test.jsx [hunk 8]
```diff
@@ -1,43 +1,63 @@
...
+    it('Should return an empty object when given an empty hostname and valid href', () => {
         const href = 'https://gitlab.com/mattermost/test/-/merge_requests/123';
         const hostname = '';
-        const expected = [''];
 
         const result = getInfoAboutLink(href, hostname);
+        expect(result).toEqual({});
+    });
+
+    it('Should return an empty object when given a valid hostname and invalid href', () => {
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/link_tooltip/link_tooltip.test.jsx:42]:
```suggestion
    it('Should return an empty object when given a valid hostname and a invalid href', () => {
```

**Pull request merged**